File: src/modules/common/connector/connectorFactory.ts
import { IConnector } from "./iconnector";
import {JsonConnector} from "./jsonConnector";
import { ConnectorType } from "../models/enums";

export class ConnectorFactory {
    public static create(type: ConnectorType): IConnector {
        switch (type) {
            case ConnectorType.Json:
                return new JsonConnector();
        }
    }
}

File: RestfulAPINew/TinyERP/TinyERP.Common/Connector/ConnectorFactory.cs
namespace TinyERP.Common.Connector
{
    public static class ConnectorFactory
    {
        public static IConnector Create(ConnectorType type)
        {
            switch (type)
            {
                case ConnectorType.Json:
                default:
                    return new JsonConnector();
            }
        }

    }
}
